src/spdx2/agent/spdx2utils.php

```diff
@@ -63,7 +63,11 @@ static public function addPrefixOnDemand($license, $spdxValidityChecker = null)
       }
       else
       {
-        return self::$prefix . $license;
+        if(!empty($license)){
+          return self::$prefix . $license;
+        }else{
+          return "NOASSERTION";
```

Comment:
Not if you add the suggested `if($license === "NOASSERTION"...` to the top of the function `addPrefixOnDemand`